internal/ethapi/api.go

```diff
@@ -1600,9 +1600,13 @@ func (s *PublicTransactionPoolAPI) GetTransactionReceipt(ctx context.Context, ha
 	}
 	receipt := receipts[index]
 
+	return ToTransactionReceipt(ctx, s.b, tx, receipt, blockHash, hash, blockNumber, index)
+}
+
+func ToTransactionReceipt(ctx context.Context, b Backend, tx *types.Transaction, receipt *types.Receipt, blockHash common.Hash, hash common.Hash, blockNumber uint64, index uint64) (map[string]interface{}, error) {
```

Comment:
Another minor suggestion: you need the chain config and the block header from the `Backend`. They can be passed to this function directly instead of the `Backend` object

Reply:
Thanks for your opinion,  I pass the backend Instead of header and chain config because the header maybe not used at this function.

```
    if !b.ChainConfig().IsLondon(bigblock) {
        fields["effectiveGasPrice"] = hexutil.Uint64(tx.GasPrice().Uint64())
    } else {
        header, err := b.HeaderByHash(ctx, blockHash)
        if err != nil {
            return nil, err
        }
        gasPrice := new(big.Int).Add(header.BaseFee, tx.EffectiveGasTipValue(header.BaseFee))
        fields["effectiveGasPrice"] = hexutil.Uint64(gasPrice.Uint64())
    }

```

Reply:
Right I noticed that but thought the header would be in cache because we fetch the block before. But I just had a look and this is not happening. The header cache doesn't get updated when getting a block from db. I'm resolving this comment.